test/controllers/declaration/RepresentativeDetailsPageControllerSpec.scala

```diff
@@ -195,6 +197,8 @@ class RepresentativeDetailsPageControllerSpec
 
       route(app, postRequest(uri, correctRepresentativeDetailsJSON)).get.futureValue
 
+      verify(mockExportsCacheService).update(any(), any())
```

Comment:
Are we happy verifying on any? Maybe some argument captors to verify what we update? If that’s tested elsewhere in this class though it’s fine

Reply:
FYI a pattern I've seen in the past which I like is:

```
def theThingThatWasUpdated: Something = {
    val captor = ArgumentCaptor.forClass(classOf[Something])
    verify(service).methodName(captor.capture)
    captor.getValue
}
```

Then in tests you can do 
```
given(service.methodName(any())).willReturn(SomeResult)

theThingThatWasUpdated shouldBe Something(...)
```

As apposed to verifying on expected values e.g. `verify(service).methodName(Something(...))` which gets a little messy with bigger objects

Reply:
This as been addressed!